### ISSUE TITLE
Missing intelligent custom data viewer component matching

### ISSUE THREAD

**Issue by tatiang** [thunkable/thunkable-issues#709]
### Platform
Thunkable X

### Steps to Reproduce
1. Create and save a new custom data viewer layout
2. Select the custom data viewer layout as a "List Item Layout"
3. Select the Data Bindings from the local data source
4. When viewing existing blocks such as "From Data Viewer List's Item Label", the blocks have errors until the same field is chosen again as "From Data Viewer List's Item Label."

### Project Link
It's a private project but I can share it with Thunkable Staff if needed.

### Expected Behavior
The "identical" components are replaced correctly in the code blocks.

### Actual Behavior
See above and video link below.
```
  STACK TRACE AND/OR SCREENSHOTS
```
This video shows the process I described above: https://drive.google.com/file/d/1Y5t327VYJjQ7-gKc8N4XSSz4yFrUjCCP/view.

**Comment by StevenPurn** [comment]
Just to make sure I understand, you're saying that if you switch layouts and the previous layout has components which share the same name as the new layout you would like us to automatically replace the components in the blocks? I think this is technically possible but I'm not sure how big of a project it would be. So I definitely can't commit to a timeline but I can at least look into whether it is feasible or not.

**Comment by tatiang** [comment]
@StevenPurn Yes, that's what I'm suggesting. I can imagine that it's not seamless or simple to enact. Thanks for considering!

And really what I'm hoping for is a Save/Save As... choice when it comes to saving a custom data viewer layout. Because each time I make a change, I have to give it a new name (which doesn't appear anywhere within the project) and I have to manually go back and delete the old version(s) of it later. Being able to edit a layout, save it as a replacement of the existing layout, and have the fields map automatically within any blocks would be AMAZING.

**Comment by StevenPurn** [comment]
You should be able to edit the layouts once created via the 'My Data Viewer Layouts' tab on the home page with all your projects. Unfortunately, once you edit the layout you still need to switch to it and so you'll run into the same issue with the blocks needing to be replaced.
